cloud/gcp/deploytf/.nitric/modules/cdn/main.tf

```diff
@@ -0,0 +1,179 @@
+        paths   = [
+          startswith(path_rule.value.base_path, "/") ? "${path_rule.value.base_path}/*" : "/${path_rule.value.base_path}/*", 
+          startswith(path_rule.value.base_path, "/") ? "${path_rule.value.base_path}" : "/${path_rule.value.base_path}"]
+        route_action {
+          url_rewrite {
+            path_prefix_rewrite = "/"
+          }
+        }
+      }
+    }
+  }
+}
+
+# Lookup the Managed Zone for the CDN Domain
+data "google_dns_managed_zone" "cdn_zone" {
+  name = var.cdn_domain.zone_name
+}
+
+# Create DNS Records for the CDN
+resource "google_dns_record_set" "cdn_dns_record" {
+  name         = endswith(var.cdn_domain.domain_name, ".") ? var.cdn_domain.domain_name : "${var.cdn_domain.domain_name}."
+  managed_zone = data.google_dns_managed_zone.cdn_zone.name
+  type         = "A"
+  rrdatas      = [google_compute_global_address.cdn_ip.address]
+  ttl = 300
+}
+
+resource "google_dns_record_set" "www_cdn_dns_record" {
+  name         = endswith(var.cdn_domain.domain_name, ".") ? "www.${var.cdn_domain.domain_name}" : "www.${var.cdn_domain.domain_name}."
+  managed_zone = data.google_dns_managed_zone.cdn_zone.name
+  type         = "A"
+  rrdatas      = [google_compute_global_address.cdn_ip.address]
+  ttl = 300
+}
+
+resource "google_certificate_manager_certificate" "cdn_cert" {
+  provider    = google-beta
+  name        = "${var.stack_id}-cdn-cert"
+  description = "Nitric stack CDN SSL certificate"
+  scope       = "DEFAULT"
+
+  managed {
+    domains = [var.cdn_domain.domain_name]
+  }
+}
+
+resource "google_certificate_manager_certificate_map" "cdn_cert_map" {
+  provider    = google-beta
+  name        = "cert-map"
+  description = "CDN Certificate Map"
+}
+
+resource "google_certificate_manager_certificate_map_entry" "cdn_cert_map_entry" {
+  name        = "cdn-cert-map-entry"
+  description = "CDN Certificate Map Entry"
+  map = google_certificate_manager_certificate_map.cdn_cert_map.name 
+  certificates = [google_certificate_manager_certificate.cdn_cert.id]
+  matcher = "PRIMARY"
+}
+
+# Create a Target HTTPS Proxy
+resource "google_compute_target_https_proxy" "https_proxy" {
+  name             = "https-proxy"
+  # ssl_certificates = [google_compute_managed_ssl_certificate.cdn_cert.self_link]
+  certificate_map = "//certificatemanager.googleapis.com/${google_certificate_manager_certificate_map.cdn_cert_map.id}"
+  url_map          = google_compute_url_map.https_url_map.self_link
+}
+
+# Create a Global Forwarding Rule
+resource "google_compute_global_forwarding_rule" "https_forwarding_rule" {
+  name        = "https-forwarding-rule"
+  ip_address  = google_compute_global_address.cdn_ip.address
+  ip_protocol = "TCP"
+  port_range  = "443"
+  target      = google_compute_target_https_proxy.https_proxy.self_link
+  # load_balancing_scheme = "EXTERNAL_MANAGED"
+}
+
+locals {
+  all_bucket_md5s = merge(
+    [
+      for bucket in values(var.website_buckets) : bucket.website_file_md5s
```

Comment:
The keys are overriding each other (same keys on subsites) prepend the subsite path to the keys to ensure they are all included or do it a different way for the triggers.